File: 1008/Main.java
import java.util.LinkedList;
import java.util.List;
import java.util.Scanner;

public class Main {
	public static void main(String[] args) {
		Scanner scanner = new Scanner(System.in);
		List<Integer> list = new LinkedList<Integer>();
		int n = scanner.nextInt();
		for(int i = 0; i < n; i++)
			list.add(scanner.nextInt());
		int time = 0;
		int lastFloor = 0;
		for (Integer i : list) {
			if (lastFloor > i) {
				time += Math.abs(lastFloor - i) * 4;
			} else {
				time += Math.abs(lastFloor - i) * 6;
			}
			time += 5;
			lastFloor = i;
		}
		
		System.out.print(time);
	}
}

File: 1008/main.cpp
#include <cstdio>

using namespace std;

int N;

int main()
{
    scanf("%d", &N);

    int last_floor = 0;
    int time = 0;

    for (int i = 0; i < N; i++)
    {
        int next_floor;
        scanf("%d", &next_floor);
        if (next_floor > last_floor)
            time += (next_floor - last_floor) * 6;
        else
            time += (last_floor - next_floor) * 4;
        time += 5;
        last_floor = next_floor;
    }

    printf("%d", time);

    return 0;
}
